router/pcap.go

```diff
@@ -1,19 +1,39 @@
 package router
 
 import (
-	"fmt"
-
 	"code.google.com/p/gopacket/pcap"
+	"fmt"
+	"net"
+	"sync"
 )
 
-type PcapIO struct {
-	handle *pcap.Handle
+type Pcap struct {
+	iface *net.Interface
+	bufSz int
+
+	// The libpcap handle for writing packets. It's possible thata
+	// single handle could be used for reading and writing, but
+	// we'd have to examine the performance implications.
+	writeHandle *pcap.Handle
+
+	// pcap handles are single-threaded, so we need to lock around
+	// uses of writeHandle.
+	writeHandleMutex sync.Mutex
```

Comment:
I see. I was looking at the final state of this PR, i.e. your 2nd explanation with `FlowOps`, which is considerably more obscure than the first.